src/main/java/teammates/common/datatransfer/FeedbackQuestionType.java

```diff
@@ -104,12 +106,79 @@ public FeedbackResponseDetails getFeedbackResponseDetailsInstance(FeedbackQuesti
         try {
             feedbackResponseDetails.extractResponseDetails(this, questionDetails, answer);
         } catch (Exception e) {
+            e.printStackTrace();
             Utils.getLogger().warning("Failed to extract response details.\n" + e.toString());
             return null;
         }
 
         return feedbackResponseDetails;
     }
+    
+    /**
+     * Returns an instance of a corresponding Feedback*ResponseDetails class
+     *
+     * @return FeedbackResponseDetails
+     */
+    public FeedbackResponseDetails getFeedbackResponseDetailsInstance(FeedbackQuestionDetails questionDetails,
+                                                                      String[] answer, Map<String, String[]> requestParameters,
+                                                                      int questionIndx, int responseIndx) {
+        FeedbackResponseDetails feedbackResponseDetails = null;
+
+        switch (this) {
+            case TEXT:
+                feedbackResponseDetails = new FeedbackTextResponseDetails();
+                break;
+            case MCQ:
+                feedbackResponseDetails = new FeedbackMcqResponseDetails();
+                answer = appendOtherOptionFlagToAnswer(answer, requestParameters, questionIndx, responseIndx);
+                break;
+            case MSQ:
+                feedbackResponseDetails = new FeedbackMsqResponseDetails();
+                break;
+            case NUMSCALE:
+                feedbackResponseDetails = new FeedbackNumericalScaleResponseDetails();
+                break;
+            case CONSTSUM:
+                feedbackResponseDetails = new FeedbackConstantSumResponseDetails();
+                break;
+            case CONTRIB:
+                feedbackResponseDetails = new FeedbackContributionResponseDetails();
+                break;
+            case RUBRIC:
+                feedbackResponseDetails = new FeedbackRubricResponseDetails();
+                break;
+            default:
+                Assumption.fail("Failed to instantiate Feedback*ResponseDetails instance for "
+                                + this.toString() + " question type.");
+                return null;
+        }
+
+        try {
+            feedbackResponseDetails.extractResponseDetails(this, questionDetails, answer);
+        } catch (Exception e) {
+            e.printStackTrace();
+            Utils.getLogger().warning("Failed to extract response details.\n" + e.toString());
+            return null;
+        }
+
+        return feedbackResponseDetails;
+    }
+
+    private String[] appendOtherOptionFlagToAnswer(String[] answer, Map<String, String[]> requestParameters,
```

Comment:
I think we should move this function into a class related to MCQ instead of having it in FeedbackQuestionType